diff --git a/dist/cli.mjs b/dist/cli.mjs
index 630e3b2e3b3f7189a6e0531916a1950275898887..8d91ba770e5c3046f9279bb1672ed4c6e062915c 100644
--- a/dist/cli.mjs
+++ b/dist/cli.mjs
@@ -12022,7 +12022,7 @@ async function runBuildCommand(app) {
         stdout: "inherit",
         stderr: "inherit",
         env: {
-          FORCE_COLOR: "true"
+          FORCE_COLOR: process.env.NO_COLOR !== undefined ? undefined : (process.env.FORCE_COLOR ?? "true")
         }
       });
     } catch (e2) {
@@ -12055,7 +12055,7 @@ async function runBuildCommand(app) {
       stderr: "inherit",
       cwd: app.rootDir,
       env: {
-        FORCE_COLOR: "true"
+        FORCE_COLOR: process.env.NO_COLOR !== undefined ? undefined : (process.env.FORCE_COLOR ?? "true")
       }
     }).then((e2) => ({
       title: "Built library modules",
@@ -12083,7 +12083,7 @@ async function runBuildCommand(app) {
       stderr: "inherit",
       cwd: app.rootDir,
       env: {
-        FORCE_COLOR: "true"
+        FORCE_COLOR: process.env.NO_COLOR !== undefined ? undefined : (process.env.FORCE_COLOR ?? "true")
       }
     }).then((e2) => ({
       title: "Built preview (ssr) modules",
@@ -12108,7 +12108,7 @@ async function runBuildCommand(app) {
       stderr: "inherit",
       cwd: app.rootDir,
       env: {
-        FORCE_COLOR: "true"
+        FORCE_COLOR: process.env.NO_COLOR !== undefined ? undefined : (process.env.FORCE_COLOR ?? "true")
       }
     }).then((e2) => ({
       title: "Built server (ssr) modules",
@@ -12132,7 +12132,7 @@ async function runBuildCommand(app) {
       stderr: "inherit",
       cwd: app.rootDir,
       env: {
-        FORCE_COLOR: "true"
+        FORCE_COLOR: process.env.NO_COLOR !== undefined ? undefined : (process.env.FORCE_COLOR ?? "true")
       }
     }).then((e2) => ({
       title: "Built static (ssg) modules",
@@ -12159,7 +12159,7 @@ async function runBuildCommand(app) {
       stderr: "inherit",
       cwd: app.rootDir,
       env: {
-        FORCE_COLOR: "true"
+        FORCE_COLOR: process.env.NO_COLOR !== undefined ? undefined : (process.env.FORCE_COLOR ?? "true")
       }
     }).then(() => ({
       title: "Lint checked"
@@ -12197,7 +12197,7 @@ async function runBuildCommand(app) {
           stderr: "inherit",
           cwd: app.rootDir,
           env: {
-            FORCE_COLOR: "true"
+            FORCE_COLOR: process.env.NO_COLOR !== undefined ? undefined : (process.env.FORCE_COLOR ?? "true")
           }
         }).catch((e2) => {
           console.log(``);
@@ -12220,7 +12220,7 @@ async function runBuildCommand(app) {
         stderr: "inherit",
         cwd: app.rootDir,
         env: {
-          FORCE_COLOR: "true"
+          FORCE_COLOR: process.env.NO_COLOR !== undefined ? undefined : (process.env.FORCE_COLOR ?? "true")
         }
       });
     } catch (e2) {
